### DISCUSSION TITLE
How to Fix: Create Self Signed CA for local soketi server

### DISCUSSION THREAD

**Discussion by whitespacecode** [soketi/pws#1110] [Q&A]
We use Soketi installed on windows computers using self signed certificate.pem and key.pem. 
(used opensll on macos `openssl x509 -req -days 365 -in csr.pem -signkey private.key -out certificate.pem`) 

First thing we noticed is that Soketi doesn't starts on https ?

```
//...
  ssl: {
    certPath: 'C:\\users\\user\\Soketi\\certificate.pem',
    keyPath: 'C:\\users\\user\\Soketi\\private.key',
    passphrase: '',
    caPath: 'C:\\users\\user\\Soketi\\certificate.pem'
  },
  userAuthenticationTimeout: 30000,
  webhooks: { batching: { enabled: false, duration: 50 } }
}
  📡 soketi initialization....
  ⚡ Initializing the HTTP API & Websockets Server...
  ⚡ Initializing the Websocket listeners and channels...
  ⚡ Initializing the HTTP webserver...
  🕵️‍♂️ Initiating metrics endpoints...

   🎉 Server is up and running!
   📡 The Websockets server is available at 127.0.0.1:6001
   🔗 The HTTP API server is available at http://127.0.0.1:6001
   🎊 The /usage endpoint is available on port 9601.
```

Then when you go to https://127.0.0.1:6001/ i get a security warning '`MOZILLA_PKIX_ERROR_SELF_SIGNED_CERT`'
![image](https://github.com/soketi/soketi/assets/37139936/60816a6a-a336-464e-8d8a-bc0ccb0bab8c)

**Comment by JanMisker** [comment]
What is your question? This is not a Soketi issue I think. Using self-signed certificates creates various obstacles in the different browsers. You could try installing the (public part of the) certificate in the trusted certificates of the browser and/or system.

**Reply by whitespacecode** [reply]
Since Soketi is globaly installed and running local on the computers for the application to work we needed signed certificates if we don't want to manually go to https://127.0.0.1:6001/ -> click advanced ->  accept.
For me it was pretty unclear how to create self-signed working certificates so we don't need to do that last step.

We recently found [mkcert](https://github.com/FiloSottile/mkcert) to create self signed certificates.
So this ticket can be closed, i would preferred a more in depth documentation or information about valid certificates.
But if that is not a Soketi issue i understand.
